src/server/models/listItem/listItem.ts

```diff
@@ -0,0 +1,343 @@
+  ListItemGetObject,
+  Point,
+} from "./../types";
+import { recordListItemEvent } from "./../audit";
+import { CovidTestSupplierListItem, LawyerListItem } from "./providers";
+import { ListItemWithAddressCountry } from "./providers/types";
+import { Prisma } from "@prisma/client";
+import { makeAddressGeoLocationString } from "server/models/listItem/geoHelpers";
+import { getChangedAddressFields } from "./providers/helpers";
+import { geoLocatePlaceByText } from "server/services/location";
+import { rawUpdateGeoLocation } from "server/models/helpers";
+
+export async function findListItemsForList(list: List): Promise<ListItem[]> {
+  try {
+    /**
+     * TODO:- should this be using prisma.listItem.findMany(..)?
+     */
+    return await prisma.$queryRaw(`SELECT
+        "ListItem".*,
+        (
+          SELECT ROW_TO_JSON(a)
+          FROM (
+            SELECT
+              "Address"."firstLine",
+              "Address"."secondLine",
+              "Address"."city",
+              "Address"."postCode",
+              (
+                SELECT ROW_TO_JSON(c)
+                FROM (
+                  SELECT name
+                  FROM "Country"
+                  WHERE "Address"."countryId" = "Country"."id"
+                ) as c
+              ) as country
+              FROM "Address"
+              WHERE "Address".id = "ListItem"."addressId"
+          ) as a
+        ) as address,
+
+        ST_X("GeoLocation"."location"::geometry) AS lat,
+        ST_Y("GeoLocation"."location"::geometry) AS long
+
+      FROM "ListItem"
+
+      INNER JOIN "Address" ON "ListItem"."addressId" = "Address".id
+      INNER JOIN "List" ON "ListItem"."listId" = "List".id
+      INNER JOIN "Country" ON "List"."countryId" = "Country".id
+      INNER JOIN "GeoLocation" ON "Address"."geoLocationId" = "GeoLocation".id
+
+      ${pgescape(`WHERE "ListItem"."type" = %L`, list.type)}
+      AND ("ListItem"."jsonData"->'metadata'->>'emailVerified')::boolean
+      AND "Country".id = ${list.countryId}
+
+      ORDER BY "ListItem"."createdAt" DESC`);
+  } catch (error) {
+    logger.error(`approveLawyer Error ${error.message}`);
+    throw new Error("Failed to approve lawyer");
+  }
+}
+
+export async function findListItemById(
+  id: string | number
+): Promise<ListItemGetObject> {
+  try {
+    return (await prisma.listItem.findUnique({
+      where: { id: Number(id) },
+      include: {
+        address: {
+          select: {
+            id: true,
+            firstLine: true,
+            secondLine: true,
+            city: true,
+            postCode: true,
+            country: {
+              select: {
+                id: true,
+                name: true,
+              },
+            },
+          },
+        },
+      },
+    })) as ListItemGetObject;
+  } catch (error) {
+    logger.error(`findListItemById Error ${error.message}`);
+    throw new Error("Failed to approve lawyer");
+  }
+}
+
+export async function togglerListItemIsApproved({
```

Comment:
consider combining this with togglerListItemIsPublished as they're very similar in nature.  If you want to keep the separate functions for easier navigation, consider specific functions that simply set the default values and call the combined function with the validation and transaction in it.

Reply:
this is moved code, but noted. would be a good target for generics.  something like `toggle<T extends PrismaQuery, K>`